test/build.py

```diff
@@ -39,66 +45,64 @@ def ensure_wasm_executable(path_to_wasm):
         sys.exit(1)
 
 
-def ensure_clean_initial_state():
-    ensure_wasm_executable(WASM_EXEC)
-
-    if os.path.exists(OUT_DIR):
-        shutil.rmtree(OUT_DIR)
-
-    ensure_dir(OUT_DIR)
-    ensure_dir(OUT_JS_DIR)
-    ensure_dir(OUT_HTML_DIR)
-
 # JS harness.
-def replace_js_harness(js_file):
-    test_func_name = os.path.basename(js_file).replace('.', '_').replace('-', '_')
+def convert_one_wast_file(inputs):
+    wast_file, js_file = inputs
+    print('Compiling {} to JS...'.format(wast_file))
+    return run(WASM_EXEC, wast_file, '-h', '-o', js_file)
 
-    content = ["(function {}() {{".format(test_func_name)]
-    with open(js_file, 'r') as f:
-        content += f.readlines()
-    content.append('})();')
+def convert_wast_to_js(out_js_dir):
+    """Compile all the wast files to JS and store the results in the JS dir."""
 
-    with open(js_file, 'w') as f:
-        f.write('\n'.join(content))
+    inputs = []
 
-def convert_wast_to_js():
-    """Compile all the wast files to JS and store the results in the JS dir."""
-    for wast_file in glob.glob(os.path.join(WAST_DIR, '*.wast')):
+    for wast_file in glob.glob(os.path.join(WAST_TESTS_DIR, '*.wast')):
         # Don't try to compile tests that are supposed to fail.
         if 'fail.wast' in wast_file:
```

Comment:
I see that, but this script doesn't call the `run.py` script which is in test/core. Are you suggesting to use it, or to implement specific logic that tests that the result is incorrect indeed? (the latter doesn't seem too valuable, because we can't specifically check that the failure happens at compile/instanciation/runtime)
Or something else?